torch/distributed/_sharded_tensor/ops/embedding.py

```diff
@@ -104,7 +104,7 @@ def sharded_embedding(types, args, kwargs, pg):
     norm_type = kwargs.get("norm_type")
```

Comment:
After this PR, the linear and elementwise ops produce sharded results, but the embedding ops still collect all the results.

For consistency, we should have a follow up PR such that embedding ops also produce sharded results.

Reply:
Put this requirement in the gh issue: https://github.com/pytorch/pytorch/issues/72022